spring-batch-core/src/main/java/org/springframework/batch/core/repository/dao/StepExecutionDao.java

```diff
@@ -86,4 +86,13 @@ default StepExecution getLastStepExecution(JobInstance jobInstance, String stepN
 	 */
 	void addStepExecutions(JobExecution jobExecution);
 
+	/**
+	 * Counts all the {@link StepExecution} for a given step name.
+	 *
+	 * @param jobInstance the parent {@link JobInstance}
+	 * @param stepName the name of the step
+	 * @since 4.2.2
+	 * @return the count of {@link StepExecution}s for a given step
+	 */
+	int countStepExecutions(JobInstance jobInstance, String stepName);
```

Comment:
This is a breaking change for anyone implementing `StepExecutionDao`, but we can make it non breaking by using a default method in the interface (Java 8 is the baseline for SB v4). Can you please add a default method that throws an `UnsupportedOperationException` (see method `getLastStepExecution`)?

Reply:
Good catch, I changed method according to `getLastStepExecution`